src/tutorial/testing.md

```diff
@@ -1,5 +1,101 @@
 # Testing
 
+Over the decades of people doing software development
+one truth has been found:
+Untested software rarely works.
+(Many people would go so far and add
+"Most tested software doesn't work either."
+But we are all optimists here, right?)
+So, to ensure that your program does what you expect it to do,
+it is wise to test it.
+
+One easy way to do that is
+to write a `README` file
+that describes what your program should do.
+And when you feel ready to make a new release,
+go through the `README` and ensure that
+the behavior is still as expected.
+You can make this a more rigorous exercise
+by also writing down how your program should react to erroneous inputs.
+
+Here's another fancy idea:
+Write that `README` before you write the code.
+
+<aside>
+
+**Aside:**
+Have a look at
+[Test-driven development](https://en.wikipedia.org/wiki/Test-driven_development) (TDD)
+if you haven't heard of it.
+
+</aside>
+
+## Automated testing
+
+Now, this is all fine and dandy,
+but doing all of this manually?
+That can take a lot of time.
+I don't know about you,
+but I really like to tell computers to do things for me.
+Let's talk about how to automate these tests.
+
+Rust has a built-in test framework,
+so let's start by writing a first test:
+
+```rust
+#[test]
+fn check_answer_validity() {
+    assert_eq!(answer(), 42);
+}
+```
+
+You can put this snippet of code in pretty much any file
+and `cargo test` will find it
+and run it.
+All the magic is done by the `#[test]` attribute.
```

Comment:
Would be great to avoid "magic" in official books/docs, there is no magic unless it's unsafe :D

Something like this could work:

>The key here is the `#[test]` attribute, it allows the build system to discover such functions and run them as tests, verifying that they don't panic.